tests/test_resources.py

```diff
@@ -147,3 +147,16 @@ def test_force_compact(tako_env):
     t1 = make_program_task(ProgramDefinition(["hostname"]), resources=rq1)
     session.submit([t1])
     session.wait_all([t1])
+
+
+def test_hw_monitoring(tako_env):
+    session = tako_env.start(
+        workers=[1], worker_start_delay=0.4, panic_on_worker_lost=False, heartbeat=500, enable_worker_hwpoll=True
+    )
+    time.sleep(0.2)
+
+    overview = session.overview()
+    cpu_percent_vec = overview["worker_overviews"][0]["hw_state"]["worker_hw_state"]["worker_cpu_usage"]["cpu_per_core_percent_usage"];
```

Comment:
Let's also check that the number of cores in the overview is the same as the number of cores on the system. For example using `multiprocessing.cpu_count()`.